### DISCUSSION TITLE
OR condition behaviour

### DISCUSSION THREAD

**Discussion by Ali-Zmn** [data-driven-forms/react-forms#1226] [Need help] [answered]
When I'm using OR condition, sometimes it needs all the fields to be touched to evaluate it.
Steps to reproduce with [Codesandbox](https://codesandbox.io/s/react-data-driven-mwdfc5?file=/src/App.jsx)
Scenario `#1` Correct behaviour
1. Enter `Bond` for the last name
2. Enter `Charles` for Father's name 
3. Now the occupation will appear as `SPY`

Scenario `#2` unexpected behaviour
1. Enter `Bond` for the last name
2. Ignore the father's name
3. Enter `Gun` for equipment 
4. No `SPY` !!!
5. Try type something in the father's name
6. Now works!

**Comment by Hyperkid123** [comment]
Seems like the condition is evaluated one cycle too late. Let create an issue.

**Comment by rvsia** [comment, marked as answer]
Hello @Ali-Zmn,

the fix is released, check here: https://github.com/data-driven-forms/react-forms/issues/1227

Please let us know, if there are any other issues (or if the issues persists!)
